src/components/Reports/GoalCalculator/SharedComponents/GoalCalculatorGrid/GoalCalculatorGrid.tsx

```diff
@@ -232,36 +259,68 @@ const GoalCalculatorGridForm: React.FC<GoalCalculatorGridFormProps> = ({
   ];
 
   return (
-    <StyledCard>
-      <StyledControlsBox>
-        <StyledAddButton
-          variant="text"
-          onClick={addExpense}
-          size="small"
-          startIcon={<AddIcon />}
-        >
-          {t('Add {{category}}', { category: categoryName })}
-        </StyledAddButton>
-        <StyledToggleBox>
-          <Switch
-            checked={directInput}
-            onChange={handleDirectInputToggle}
+    <>
+      <StyledBox>
+        <Typography variant="h6" component="span" sx={{ mr: 2 }}>
+          {categoryName}
+        </Typography>
+        <StyledButtonGroup>
+          <StyledModeButton
+            variant="outlined"
             size="small"
-          />
-
-          <StyledDirectInputText variant="button">
-            {t('Direct Input')}
-          </StyledDirectInputText>
-        </StyledToggleBox>
-      </StyledControlsBox>
+            isActive={directInput}
+            onClick={() => setDirectInput(true)}
+            startIcon={<FunctionsIcon />}
+          >
+            {t('Lump Sum')}
+          </StyledModeButton>
+          <StyledModeButton
+            variant="outlined"
+            size="small"
+            isActive={!directInput}
+            onClick={() => setDirectInput(false)}
+            startIcon={<ViewHeadlineIcon />}
+          >
+            {t('Line Item')}
+          </StyledModeButton>
+        </StyledButtonGroup>
+      </StyledBox>
+      <StyledCard>
+        {directInput ? (
+          <Box sx={{ p: 2 }}>
+            <TextField
+              fullWidth
+              size="small"
+              label={t('Total')}
+              type="number"
+              value={values.lumpSumAmount}
+              onChange={(e) => setFieldValue('lumpSumAmount', e.target.value)}
+              sx={{ mb: 2 }}
+            />
+          </Box>
+        ) : (
+          <>
+            <StyledControlsBox>
+              <StyledAddButton
+                variant="text"
+                onClick={addExpense}
+                size="small"
+                startIcon={<AddIcon />}
+              >
+                {t('Add Expense')}
```

Comment:
I wonder if this should be "Add Line Item"? This grid will be used on things that aren't expenses, like one-time goals and additional income.